### PR TITLE
Fix #1566 - Improve Cloudflare disable notice

### LINKED ISSUE #1566: Proposal: Improve notice when Cloudflare plugin is detected
When the official Cloudflare plugin is detected, we show our generic deactivation warning:

> The following plugins are not compatible with this plugin and may cause unexpected results:

But we still get tickets about this - it's not clear to customers why they're incompatible or if there will be some other problem if they deactivate one or the other, etc. 

I think we could improve the notice to explain a bit further. Along the lines of:  
```
The Cloudflare plugin is not required because the WP Rocket Cloudflare add-on has the same functionality. 
```

### PR DESCRIPTION
**Pull request by crystinutzaa**
Improve notice when Cloudflare plugin is detected

Edited the current notice view and plugin name is in bold now with an explainer text beside it. Increased the width of .rocket-plugins-error li from 280px to 100% to allow the explainer text to fit inside:
https://www.screencast.com/t/0hKXgvhfGc6Y

Based on:
3.4.1 WP Rocket Minor Version doc

### CODE REVIEW

File: inc/admin/ui/notices.php [hunk 1]
```diff
@@ -92,7 +92,8 @@ function rocket_warning_plugin_modification() {
  * @since 1.3.0
  */
 function rocket_plugins_to_deactivate() {
-	$plugins = [];
+	$plugins 			  = [];
+	$plugins_explanations = [];
```
Comment (Tabrisrp) [inline inc/admin/ui/notices.php]: Alignement to fix, `=` should be on the same column

File: inc/admin/ui/notices.php [hunk 2]
```diff
@@ -166,7 +167,8 @@ function rocket_plugins_to_deactivate() {
 	}
 
 	if ( get_rocket_option( 'do_cloudflare' ) ) {
-		$plugins['cloudflare'] = 'cloudflare/cloudflare.php';
+		$plugins['cloudflare'] 			   	= 'cloudflare/cloudflare.php';
+		$plugins_explanations['cloudflare'] = __( 'The Cloudflare plugin is not required because the WP Rocket Cloudflare add-on has the same functionality.', 'rocket' );
```
Comment (Tabrisrp) [inline inc/admin/ui/notices.php]:
```suggestion
		$plugins_explanations['cloudflare'] = __( 'WP Rocket Cloudflare Add-on provides similar functionalities. They can not be active at the same time.', 'rocket' );
```

File: inc/admin/ui/notices.php [hunk 3]
```diff
@@ -194,9 +196,9 @@ function rocket_plugins_to_deactivate() {
 
 		$warning .= '<ul class="rocket-plugins-error">';
 
-		foreach ( $plugins as $plugin ) {
+		foreach ( $plugins as $k => $plugin ) {
 			$plugin_data = get_plugin_data( WP_PLUGIN_DIR . DIRECTORY_SEPARATOR . $plugin );
-			$warning    .= '<li>' . $plugin_data['Name'] . '</span> <a href="' . wp_nonce_url( admin_url( 'admin-post.php?action=deactivate_plugin&plugin=' . rawurlencode( $plugin ) ), 'deactivate_plugin' ) . '" class="button-secondary alignright">' . __( 'Deactivate', 'rocket' ) . '</a></li>';
+			$warning    .= '<li><b>' . $plugin_data['Name'] . '</b>'. ( isset($plugins_explanations[$k]) ? ' - ' . $plugins_explanations[$k] : '' ) . '</span> <a href="' . wp_nonce_url( admin_url( 'admin-post.php?action=deactivate_plugin&plugin=' . rawurlencode( $plugin ) ), 'deactivate_plugin' ) . '" class="button-secondary alignright">' . __( 'Deactivate', 'rocket' ) . '</a></li>';
```
Comment (Tabrisrp) [inline inc/admin/ui/notices.php]:
```suggestion
			$warning    .= '<li><b>' . $plugin_data['Name'] . '</b>'. ( isset( $plugins_explanations[$k] ) ? ' - ' . $plugins_explanations[$k] : '' ) . '</span> <a href="' . wp_nonce_url( admin_url( 'admin-post.php?action=deactivate_plugin&plugin=' . rawurlencode( $plugin ) ), 'deactivate_plugin' ) . '" class="button-secondary alignright">' . __( 'Deactivate', 'rocket' ) . '</a></li>';
```

File: inc/admin/ui/notices.php [hunk 4]
```diff
@@ -92,7 +92,8 @@ function rocket_warning_plugin_modification() {
  * @since 1.3.0
  */
 function rocket_plugins_to_deactivate() {
-	$plugins = [];
+	$plugins 			  = [];
```
Comment (Tabrisrp) [inline inc/admin/ui/notices.php]:
```suggestion
	$plugins                       = [];
```

Comment (crystinutzaa) [reply inc/admin/ui/notices.php]: Please recheck

**Pull request merged**